docs/IntegrationGuide.md

```diff
@@ -1,95 +1,306 @@
+
+#### Validate payment amount
+There must be at least one CTP payment object of type PayPal Plus (`#Payment#paymentMethodInfo#paymentInterface` = `PAYPAL_PLUS`) where `Payment#amountPlanned` matches current cart's total amount and currency.
+
+#### Validate payment transaction
+Cart's payment counts as successful if there is at least one payment object with successful (`Payment#Transaction#state`=`Success`) payment transaction of type `Charge` and for the same payment object there is no other successful (`#Payment#Transaction#state`=`Success`) transactions of **other** type than `Charge` and `#Transaction#amount` greater than `0`.
+
+### Possible edge cases
+1. First case:
+    1. User inputs an address Berlin, Germany
+    1. User clicks on Continue and is redirected to Paypal payment page with this address.
+    1. In a different tab, the same user changes his address to Paris, France.
+    1. The user confirms the payment in the first tab and is redirected back to shop
+    1. Paypal has the address of Berlin, but the shop will deliver the goods to Paris.
+
+    **Possible solution:** the backend calls `patch/payments` endpoint every time user changes it.
+
+1. Second case:
+    1. User has cart with item1=10€
+    1. User is redirected to Paypal payment page
+    1. In a different tab, user changes his cart to e.g. item2=20€
+    1. User completes the payment in the first tab and is redirected back to shop
+    1. Paypal approves the payment for 10€, but the real total amount of sold (shipped) items has changed to 20€
+
+    **Possible solution:** the backend has to compare total amount of the payment and total amount of payment's cart before calling `execute/payments` endpoint.
+     In case of differences, the whole payment process must be restarted.
+
+### Bad practice
+- Never delete or un-assign created payment objects **during checkout** from the cart. If required &mdash; clean up  unused/obsolete payment objects by another asynchronous process instead.
+
+## PayPal integration service API
+
+### HTTP Responses
+All endpoints accept and return data as JSON.
+
+1. Return HTTP codes on `create/payments` endpoint URL:
+- **201**: successfully created payment in PayPal and CTP updated with `approvalUrl` as custom field
+
+1. Return HTTP codes on `execute/payments` endpoint URL:
+- **201**: successfully executed payment in PayPal, created transaction in CTP
+
+1. Return HTTP codes on `paypalplus/payments/${paypalPaymentId}` endpoint URL:
+- **200**: Paypal payment found and returned as JSON response body
+
+1. Common error codes
+- **404**: resource not found by the supplied UUID/ID
+- **400**: required request parameters are missing or wrong
+- **503**: any exception which implies that request can be safely retried with the same parameters/payload again
+- **500**: unexpected/not handled exceptions
```

Comment:
> 1.

Wrong ordered list formatting. All of them parsed as "1.", instead of incrementing. I recommend here to format like bullet points with intension:
- Return HTTP codes on `create/payments` endpoint URL:
  - **201**: successfully created payment in PayPal and CTP updated with `approvalUrl` as custom field
 
- Return HTTP codes on `execute/payments` endpoint URL:
  - **201**: successfully executed payment in PayPal, created transaction in CTP
 
- Return HTTP codes on `paypalplus/payments/${paypalPaymentId}` endpoint URL:
  - **200**: Paypal payment found and returned as JSON response body

- Common error codes
  - **404**: resource not found by the supplied UUID/ID
  - **400**: required request parameters are missing or wrong
  - **503**: any exception which implies that request can be safely retried with the same parameters/payload again
  - **500**: unexpected/not handled exceptions